src/components/contextual/pages/pool/invest/InvestPage.vue

```diff
@@ -0,0 +1,195 @@
+      // Swap input token between Native and Wrapped Native assets
+      useNativeAsset.value = isNativeAsset;
+    }
+
+    // SetTimeout hack.
+    // Race condition when switching between 'native'/'wrapped native' assets,
+    // and then setting the new input value
+    setTimeout(() => {
+      // Set the input value to max token balance for lecacy invest state
+      const indexOfAsset = indexOfAddress(tokenAddresses.value, tokenAddress);
+
+      if (indexOfAsset >= 0) {
+        amounts.value[indexOfAsset] = maxBalance;
+      }
+    }, 50);
+  }
+}
+
+/**
+ * CALLBACKS
+ */
+onBeforeMount(async () => {
+  await forChange(loadingPool, false);
+
+  if (pool.value && isDeepPool.value) {
+    // Initialise SOR for batch swap queries
+    sorReady.value = await sor.fetchPools();
+  } else {
+    sorReady.value = true;
+  }
+});
+</script>
+
+<template>
+  <!-- TODO: Move staking provider somewhere else. -->
+  <StakingProvider :poolAddress="pool?.address">
+    <div class="invest-page-layout-grid">
+      <div v-if="!upToLargeBreakpoint" class="col-span-5">
+        <BalLoadingBlock
+          v-if="loadingPool || !transfersAllowed || !pool"
+          class="h-64"
+        />
+        <div v-else>
+          <MyWallet
+            includeNativeAsset
+            :excludedTokens="excludedTokens"
+            :pool="pool"
+            @click:asset="handleMyWalletTokenClick"
+          />
+        </div>
+      </div>
+
+      <div class="col-span-7">
+        <BalLoadingBlock
+          v-if="loadingPool || !transfersAllowed || !sorReady || !pool"
+          class="h-96"
+        />
+        <BalCard v-else shadow="xl" exposeOverflow noBorder>
+          <template #header>
+            <div class="w-full">
+              <div class="text-xs leading-none text-secondary">
+                {{ network.chainName }}
+              </div>
+              <div class="flex justify-between items-center">
+                <h4>{{ $t('investInPool') }}</h4>
+                <TradeSettingsPopover :context="TradeSettingsContext.invest" />
+              </div>
+              <BalTabs
+                v-if="isDeepPool"
+                v-model="activeTab"
+                :tabs="tabs"
+                class="p-0 m-0 -mb-px whitespace-nowrap"
+                noPad
+              />
+            </div>
+          </template>
+          <template v-if="isDeepPool">
+            <InvestFormV2 :pool="pool" />
+          </template>
+          <template v-else>
+            <InvestForm :pool="pool" />
+          </template>
+        </BalCard>
+      </div>
+
+      <BalAccordion
+        v-if="upToLargeBreakpoint"
+        class="mt-4"
+        :sections="[
+          {
+            title: poolSupportsGeneralisedJoin
+              ? `${$t('myWallet2')} ${
+                  hasNativeBalance ? `${nativeBalance} ${nativeCurrency}` : ''
+                }`
+              : $t('poolTransfer.myWalletTokensCard.title'),
```

Comment:
I'm not a fan of logic like this in the templates, it makes it hard to parse, please extract to a computed property.

Reply:
Done